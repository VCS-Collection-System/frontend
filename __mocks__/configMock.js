export const getConfig = (URL, configUrl, config) => {
  if(URL.indexOf(configUrl) > -1 ) {
    const country = URL.substring(configUrl.length);
    const countryConfig = config[country] || null;
    return Promise.resolve({data: countryConfig});
  } else {
    return Promise.reject({data: null});
  }
}