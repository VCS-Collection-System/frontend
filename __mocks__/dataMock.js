export const employeeInfo = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth":"2021-10-07",
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const attachmentObject = {
  "originalFileName":"blob",
  "size":100,
  "contentType":"image/png",
  "s3BucketName":"mockBucket",
  "s3UUID":"mockUUID"
}

export const vaxHistory = {
  "id":0,
  "employee": employeeInfo,
  "submissionDate":"2021-10-04T15:12:14.875",
  "review":{
     "reviewerEmployeeId":"z2",
     "reviewerNotes":"",
     "reviewDate":"2021-10-05T14:04:29.296",
     "reviewOutcome":"ACCEPTED",
     "rejectReason":""
  },
  "submittedBy":null,
  "autoApproved":false,
  "lastUpdatedTime":null,
  "lastUpdatedBy":null,
  "vaccineBrand":"MODERNA",
  "vaccineAdministrationDate":"2021-10-04",
  "fullVaccinatedFlag":false,
  "vaccineShotNumber":2,
  "documentType":"VaccineDocument",
  "attachment": {}
}

const taskEmployeeObject = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth": {
    "year": "2021",
    "monthValue": "12",
    "dayOfMonth": "16"
  },
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const mockTask = {
  "task-input-data": {
    "documentList": {
      "com.redhat.vcs.model.VaccineDocumentList": {
        "documents": [ {
          "com.redhat.vcs.model.VaccineCardDocument": {
            "employee": {
              "com.redhat.vcs.model.Employee": taskEmployeeObject
            },
            "vaccineBrand": {
              "com.redhat.vcs.model.VaccineBrand": "MODERNA"
            },
            "attachment": {
              "com.redhat.vcs.model.Attachment": attachmentObject
            },
            "vaccineAdministrationDate": {
              "year": "2021",
              "monthValue": "12",
              "dayOfMonth": "16"
            },
            "lotNumber": "mockLot"
          } }
        }
      }
    }
  }
}
