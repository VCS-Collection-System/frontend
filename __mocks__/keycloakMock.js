module.exports = {
    token: "bogustoken", 
    authenticated: false,
    loadUserProfile: jest.fn( ()=> {return Promise.resolve({
        attributes: {
            workforceid: ["mockId"]
        }
    })} ),
    login: jest.fn(),
    logout: jest.fn(),
    hasRealmRole: jest.fn(),
    profile: {
        firstName: "mockFirst",
        lastName: "mockLast",
        email: "mock@email.com",
        attributes: {
            workforceid: ["mockId"],
            agency: ["USA"]
        }
    }
};
