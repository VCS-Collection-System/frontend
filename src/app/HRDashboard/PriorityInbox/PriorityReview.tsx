import * as React from 'react';
import { useEffect } from 'react';
import { 
  Button,
  Card, 
  CardBody, 
  CardFooter,
  Grid,
  Flex,
  FlexItem,
  GridItem
} from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import { CheckCircleIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';
import { useState } from 'react';
import authaxios from "@app/utils/axiosInterceptor";
import EmployeeInfoCard from '@app/Components/EmployeeInfoCard';
import ReviewInfoCard from '@app/Components/ReviewInfoCard';
import { 
  ATTACHMENT_URL,
  EMPLOYEE_URL,
  PRIORITY_CONFIRM_URL
} from "@app/utils/constants"; 

const PriorityReview: React.FunctionComponent = () => {

  const history = useHistory();
  const [covidTestResult, setCovidTestResult] = useState("");
  const [covidTestDate, setCovidTestDate] = useState("");
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");
  const [employeeDOB, setEmployeeDOB] = useState("");
  const [employeeEmail, setEmployeeEmail] = useState("");
  const [employeeAgency, setEmployeeAgency] = useState("");
  const [employeeDivision, setEmployeeDivision] = useState("");
  const [priorityId, setPriorityId] = useState(""); 
  const [attachment, setAttachment] = useState("");
  const [attachmentType, setAttachmentType] = useState(""); 
  const [attachmentSize, setAttachmentSize] = useState(0);  

  useEffect(() => {
    const sessionData = sessionStorage.getItem("priorityData") || "";
    const priorityData = JSON.parse(sessionData);
    const personId = priorityData['employeeId'];
    const resAttachment = priorityData['attachment'];
    setAttachmentSize(resAttachment.size);

    if(priorityData['submissionType'] === "recovery") {
      setCovidTestResult("RECOVERY");
    } else {
      setCovidTestResult("POSITIVE")
    }
    setCovidTestDate(priorityData['covidTestDate']); 
    setPriorityId(priorityData['id']); 

    const headers = {
      'Accept': 'application/json'
    }

    authaxios.get(EMPLOYEE_URL + personId, {
      headers: headers
    })
    .then(res => {
      const personData = res['data'];
      setEmployeeId(personData['employeeId']);
      setEmployeeName(personData['firstName'] + " " + personData['lastName']);
      setEmployeeDOB(personData['dateOfBirth']);
      setEmployeeEmail(personData['alternateEmail'] ? personData['alternateEmail'] : (personData['email'] ? personData['email'] : "Not Provided")); 
      setEmployeeAgency(personData['agencyCode']);
      setEmployeeDivision(personData['divisionCode']); 
    })

    authaxios.post(ATTACHMENT_URL, resAttachment, {
      responseType: 'arraybuffer'
    })
    .then(res => {
      setAttachment(res['data']);
      setAttachmentType(res.headers['content-type']);
    })

  }, []);

  function handleConfirm() {

    const headers = {
      'Accept': 'application/json'
    }

    authaxios.delete(PRIORITY_CONFIRM_URL + priorityId, {
      headers: headers
    })
    .then(() => {
      history.push('/hrdashboard');
    })
  }

  return (
    <Card isRounded>
      <CardWithTitle title="HR Review" info="Review COVID Results"/>
      <CardBody>
        <Grid hasGutter>
          <GridItem rowSpan={2} span={6}> 
          {
            attachment && attachmentSize === 0 &&
            "No image"
          }
          {
            attachment && attachmentType != 'application/pdf' && attachmentSize > 0 &&
            <img src={URL.createObjectURL(new Blob([attachment]))} /> 
          }
          {
            attachment && attachmentType == 'application/pdf' && attachmentSize > 0 &&
            <Button
              icon={<ExternalLinkAltIcon />}
              variant="link"
              component="a"
              href={URL.createObjectURL(new Blob([attachment], {type: 'application/pdf'}))}
              target="_blank"
            >
              Test Result PDF
            </Button>
          }
          </GridItem>
          <GridItem span={6}> 
            <ReviewInfoCard
              reviewType="TEST"
              covidTestResult={covidTestResult}
              covidTestDate={covidTestDate}
            />
            <EmployeeInfoCard 
              employeeId={employeeId}
              employeeName={employeeName}
              employeeDOB={employeeDOB}
              employeeEmail={employeeEmail}
              employeeAgency={employeeAgency}
              employeeDivision={employeeDivision}
            />
          </GridItem>
        </Grid>
      </CardBody>
      <CardFooter>
        <Flex>
          <FlexItem align={{ default: "alignLeft" }}>
            <Button variant="plain" onClick={() => history.push("/hrdashboard")}>
              Back
            </Button>
          </FlexItem>
          <FlexItem align={{ default: "alignRight" }}>
            <Button icon={<CheckCircleIcon />} variant="primary" onClick={handleConfirm}>
              Confirm
            </Button>
          </FlexItem>
        </Flex>
      </CardFooter>
    </Card>
  );
}

export { PriorityReview };