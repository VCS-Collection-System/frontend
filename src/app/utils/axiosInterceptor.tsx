import axios from "axios";

const axiosInterceptor = axios.create({})

axiosInterceptor.interceptors.request.use(
    config => {
        config.headers['Authorization'] = 'Bearer ' + sessionStorage.getItem("token");
        return config;
    },
    error => {
        Promise.reject(error)
    });
export default axiosInterceptor;