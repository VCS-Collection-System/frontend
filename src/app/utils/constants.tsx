declare global {  
  interface Window {
    REACT_APP_PAM_URL_BASE: string;
    REACT_APP_PAM_CONTAINER_ID: string;
    REACT_APP_KEYCLOAK_COUNTRY: string;
    REACT_APP_KEYCLOAK_EMAIL: string;
    REACT_APP_KEYCLOAK_FIRST_NAME: string;
    REACT_APP_KEYCLOAK_ID: string;
    REACT_APP_KEYCLOAK_LAST_NAME: string;
    REACT_APP_MAINTENANCE_BANNER: boolean;
    REACT_APP_MAINTENANCE_BANNER_TEXT: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_FEATURE_FLAGS: Array<Record<string, any>>;
    REACT_APP_MAX_DOSES: Record<string, number>;
    REACT_APP_CONSENT_LABEL: string;
    REACT_APP_ALTERNATE_CONSENT_LABEL: string;
    REACT_APP_BLOCKED_COUNTRIES: Array<string>;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_GLOBAL_CONFIG: Record<string, any>;
  }
}

// PAM urls
export const PAM_BASE_URL = window.REACT_APP_PAM_URL_BASE; 
export const SUBMIT_VAX_URL = PAM_BASE_URL + '/attestation/vax';
export const SUBMIT_TEST_URL = PAM_BASE_URL + '/attestation/covid-test-result';
export const PAM_TASK_URL = PAM_BASE_URL + '/tasks/';
export const RESERVED_TASK_URL = PAM_TASK_URL + 'owners?status=Reserved&user=';
export const INPROGRESS_TASK_URL = PAM_TASK_URL + 'owners?status=InProgress&user=';
export const ALL_TASK_URL = PAM_TASK_URL + 'owners?status=Ready&pageSize=200';
export const PRIORITY_URL = PAM_BASE_URL + '/query/priority-results';
export const PRIORITY_CONFIRM_URL = PAM_BASE_URL + '/query/priority-result/';
export const ATTACHMENT_URL = PAM_BASE_URL + '/attestation/attachment';
export const EMPLOYEE_URL = PAM_BASE_URL + '/query/employee/';
export const EMPLOYEE_NAME_URL = PAM_BASE_URL + '/query/employee-name/';
export const EMPLOYEE_VAX_HISTORY_URL = PAM_BASE_URL + '/query/vax-document/';
export const EMPLOYEE_TEST_HISTORY_URL = PAM_BASE_URL + '/query/test-result-document/';
export const VAX_TASK_DOCUMENT_URL = PAM_BASE_URL + '/query/task/document/';
export const DELETE_VAX_URL = PAM_BASE_URL + '/attestation/delete-vax-record/';
export const CONFIG_URL = PAM_BASE_URL + '/query/country-configs/';
export const SAVE_CONFIG_URL = PAM_BASE_URL + '/attestation/country-configs';

// List of countries allow to use this app
export const BLOCKED_COUNTRIES = window.REACT_APP_BLOCKED_COUNTRIES || []

export const KONAMI_CODE = [ "ArrowUp", "ArrowUp", "ArrowDown", "ArrowDown", "ArrowLeft", "ArrowRight", "ArrowLeft", "ArrowRight", "KeyB", "KeyA"];

// Maintenance Banner
export const MAINTENANCE_BANNER = window.REACT_APP_MAINTENANCE_BANNER || false
export const MAINTENANCE_BANNER_TEXT = window.REACT_APP_MAINTENANCE_BANNER_TEXT || ""

// Keycloak claims
export const KEYCLOAK_COUNTRY = window.REACT_APP_KEYCLOAK_COUNTRY
export const KEYCLOAK_EMAIL = window.REACT_APP_KEYCLOAK_EMAIL
export const KEYCLOAK_FIRST_NAME = window.REACT_APP_KEYCLOAK_FIRST_NAME
export const KEYCLOAK_ID = window.REACT_APP_KEYCLOAK_ID
export const KEYCLOAK_LAST_NAME = window.REACT_APP_KEYCLOAK_LAST_NAME
export const FEATURE_FLAGS = window.REACT_APP_FEATURE_FLAGS || []

// Maximum doses allowed by vaccine manufacturer
export const MAX_DOSES = window.REACT_APP_MAX_DOSES || {}

export const CONSENT_LABEL = window.REACT_APP_CONSENT_LABEL
export const ALTERNATE_CONSENT_LABEL = window.REACT_APP_ALTERNATE_CONSENT_LABEL

// Global country config
export const GLOBAL_CONFIG = window.REACT_APP_GLOBAL_CONFIG || {};

