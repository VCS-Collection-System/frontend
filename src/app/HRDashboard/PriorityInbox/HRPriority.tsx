import React, { useState } from 'react';
import { Menu, MenuContent, MenuGroup, MenuList, MenuItem, Card } from '@patternfly/react-core';
import { useHistory } from 'react-router-dom';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useEffect } from 'react';
import authaxios from "@app/utils/axiosInterceptor";
import { dateConvert } from "@app/utils/utils";
import { PRIORITY_URL } from "@app/utils/constants"; 

const HRPriority: React.FunctionComponent = () => {
  const [activeItem, setActiveItem] = useState('0');
  const [inboxList, setInboxList] = useState([]);
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const [positiveList, setPositiveList] = useState<Record<string, any>[]>([]);
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const [recoveryList, setRecoveryList] = useState<Record<string, any>[]>([]);
  const history = useHistory();

  function onSelect(event, itemId) {
    sessionStorage.setItem("priorityData", JSON.stringify(inboxList.find(item => item['id'] === itemId)));
    setActiveItem(itemId);
    history.push('/hrdashboard/priorityreview');
  }

  useEffect(() => {
    const headers = {
      'Accept': 'application/json'
    }

    authaxios.get(PRIORITY_URL, {
      headers: headers
    })
    .then(res => {
      setInboxList(res['data']);
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const positiveItems: Record<string, any>[] = [];
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const recoveryItems: Record<string, any>[] = [];
      res['data'].forEach(item => {
        if(item.submissionType === "recovery") {
          recoveryItems.push(item);
        } else {
          positiveItems.push(item);
        }
      });
      setPositiveList(positiveItems);
      setRecoveryList(recoveryItems);
    })
    .catch((res) => {res})
  }, []);

  return (
    <Card isRounded className="warningText">
      <CardWithTitle title="Priority Inbox" info="Select an item below to review" />
      <Menu onSelect={onSelect} activeItemId={activeItem}>
        <MenuContent menuHeight="350px">
          <MenuGroup label="Positive Tests">
              <MenuList id="positive-test-list" className="indentList">
              {
                positiveList.length == 0 &&
                <MenuItem key="empty" itemId="empty" isDisabled>
                  [No Positive Tests]
                </MenuItem>
              }
              {
                positiveList.length > 0 &&
                  positiveList.map((item, index) => (
                    <MenuItem key={index} description={"Test Date: " + dateConvert(item['covidTestDate'])} itemId={item['id']}>
                      {item['employeeId']}
                    </MenuItem>
                  ))
              }
            </MenuList>
          </MenuGroup>
          <MenuGroup label="Certificates of Recovery">
              <MenuList id="recovery-list" className="indentList">
              {
                recoveryList.length == 0 &&
                <MenuItem key="empty" itemId="empty" isDisabled>
                  [No Certificates of Recovery]
                </MenuItem>
              }
              {
                recoveryList.length > 0 &&
                  recoveryList.map((item, index) => (
                    <MenuItem key={index} description={"Test Date: " + dateConvert(item['covidTestDate'])} itemId={item['id']}>
                      {item['employeeId']}
                    </MenuItem>
                  ))
              }
            </MenuList>
          </MenuGroup>
        </MenuContent>
      </Menu>
    </Card>
  );
};

export { HRPriority };