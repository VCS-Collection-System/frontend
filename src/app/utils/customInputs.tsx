import { useEffect, useState } from "react";
import { KONAMI_CODE } from './constants'

const useInputEvent = () => {
  const [key, setKey] = useState(null);
  useEffect(() => {
    const keyDownHandler = ({ code }) => setKey(code);
    const keyUpHandler = () => setKey(null);
    window.addEventListener('keydown', keyDownHandler);
    window.addEventListener('keyup', keyUpHandler);
    return () => {
      window.removeEventListener("keydown", keyDownHandler);
      window.removeEventListener("keyup", keyUpHandler)
    }
  }, []);
  return key;
}

const useSecretCode = (secretCode) => {
  const [count, setCount] = useState(0);
  const [success, setSuccess] = useState(false);
  const key = useInputEvent();
  
  useEffect(() => {
    if (key == null) return;
    if (key !== secretCode[count]) {
      setCount(0);
      return;
    }
    
    setCount((state) => state + 1);
    if (count + 1 === secretCode.length) {
      setSuccess(!success);
      setCount(0);
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [key]); 
  
  return success;
};

export const useDarkMode = () => {
  return useSecretCode(KONAMI_CODE);
};
