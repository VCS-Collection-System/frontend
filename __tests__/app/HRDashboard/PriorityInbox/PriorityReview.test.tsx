import React from 'react';
import { PriorityReview } from '../../../../src/app/HRDashboard/PriorityInbox/PriorityReview';
import mockKeycloak from '../../../../__mocks__/keycloakMock'
import { attachmentObject, mockTask } from '../../../../__mocks__/dataMock'
import { act } from 'react-dom/test-utils';
import { ATTACHMENT_URL, EMPLOYEE_URL, PRIORITY_CONFIRM_URL } from '../../../../src/app/utils/constants'
import { fireEvent, render, screen } from '@testing-library/react'
import reactRouterDom from 'react-router-dom'

jest.mock('react-router-dom')
const mockPush = jest.fn()
reactRouterDom.useHistory = jest.fn().mockReturnValue({push: mockPush})

jest.mock('@react-keycloak/web', () => ({
  useKeycloak: jest.fn( ()=>({keycloak: mockKeycloak}))
})) 

jest.mock("@app/utils/constants", () => ({
  KEYCLOAK_FIRST_NAME: "firstName",
  KEYCLOAK_LAST_NAME: "lastName",
  KEYCLOAK_EMAIL: "email",
  KEYCLOAK_COUNTRY: "attributes.agency[0]",
  KEYCLOAK_ID: "attributes.workforceid[0]",
  APPROVER_ROLES: ["mock-approver"],
  FEATURE_FLAGS: ["fakeFeature"],
  ATTACHMENT_URL: "attachment/",
  EMPLOYEE_URL: "employee/",
  PRIORITY_CONFIRM_URL: "priorityconfirm/",
  GLOBAL_CONFIG: {"default": {
    "allowPdf": false,
    "dateOfBirth": true,
    "consentCheckbox": false,
    "covidTest": false,
    "customInput": true,
    "hrdashboard": true,
    "hrsearch": true,
    "proofType": {"cdc": false, "divoc": false, "greenPass": false},
    "recovery": false,
    "fakeFeature": true
  }}
}))

jest.mock('@app/utils/axiosInterceptor', () => ({
  get: (URL) => {
    let returnData; 

    if(URL === EMPLOYEE_URL + "4200000") {
      returnData = {data: {
        employeeId: 4200000,
        firstName: "Mock",
        lastName: "Person",
        dateOfBirth: "1990-07-04",
        email: "mock@email.com"
      }}; 
      return Promise.resolve(returnData);
    }
    return Promise.resolve(returnData)
  },
  post: (URL, props) => {
    let returnData; 

    if(URL === ATTACHMENT_URL && props.originalFileName === 'mockPNG'){
      returnData = {data: {}, headers: {"content-type": "application/png"}};
    }
    else if(URL === ATTACHMENT_URL){
      returnData = {data: {}, headers: {}};
    }
    return Promise.resolve(returnData)
  },
  delete: (URL) => {
    let returnData; 
    return Promise.resolve(returnData)
  }
}))

beforeEach(() => {
  const data = {
    id: 1,
    employeeId: 4200000,
    submissionType: "positiveTest",
    covidTestDate: "2022-01-01",
    attachment: {
      originalFileName: "mockPNG",
      size: 0
    }
  }

  sessionStorage.setItem("priorityData", JSON.stringify(data));
})

afterEach(() => {
  sessionStorage.clear(); 
})

afterEach(() => {
  sessionStorage.clear(); 
})

describe('<PriorityReview />', ()=> {
  it('should be defined', () => {
    expect(PriorityReview).toBeDefined()
  })

  it("should match existing snapshot", async () => {
    await act(async () => {
      const { asFragment } = render(
        <PriorityReview />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it("should display employee information", async () => {
    await act(async () => {
      render(<PriorityReview />)
    })

    expect(screen.getByDisplayValue('4200000')).toBeTruthy();
    expect(screen.getByDisplayValue('Mock Person')).toBeTruthy();
    expect(screen.getByDisplayValue('07-04-1990')).toBeTruthy();
    expect(screen.getByDisplayValue('mock@email.com')).toBeTruthy();
  })

  it("should display information from positive test", async () => {
    await act(async () => {
      render(<PriorityReview />)
    })

    expect(screen.getByDisplayValue('POSITIVE')).toBeTruthy();
    expect(screen.getByDisplayValue('01-01-2022')).toBeTruthy();
  })

  it("should display information from a certificate of recovery", async () => {
    const data = {
      id: 1,
      employeeId: 4200000,
      submissionType: "recovery",
      covidTestDate: "2022-02-01",
      attachment: {
        originalFileName: "mockPNG",
        size: 0
      }
    }

    sessionStorage.setItem("priorityData", JSON.stringify(data));

    await act(async () => {
      render(<PriorityReview />)
    })

    expect(screen.getByDisplayValue('RECOVERY')).toBeTruthy();
    expect(screen.getByDisplayValue('02-01-2022')).toBeTruthy();
  })

  it("should be able to confirm", async () => {
    await act(async () => {
      render(<PriorityReview />)
    })

    const confirmButton = screen.getByRole('button', {name: /Confirm/i})

    await act(async () => {
      fireEvent.click(confirmButton)
    })

    expect(mockPush).toBeCalledTimes(1);
  })

  it("should return to hr dashboard when back button is clicked", async () => {
    await act(async () => {
      render(<PriorityReview />)
    })

    const backButton = screen.getByRole('button', {name: /Back/i})

    await act(async () => {
      fireEvent.click(backButton)
    })

    expect(mockPush).toBeCalledTimes(1)
    expect(mockPush).toBeCalledWith('/hrdashboard')
  })

})