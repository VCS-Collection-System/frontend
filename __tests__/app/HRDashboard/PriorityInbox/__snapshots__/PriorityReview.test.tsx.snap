// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriorityReview /> should match existing snapshot 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card pf-m-rounded"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <p
        class=""
        data-pf-content="true"
      >
        HR Review
      </p>
      <div
        class="pf-c-content"
      >
        <small
          class=""
          data-pf-content="true"
        >
          Review COVID Results
        </small>
      </div>
    </div>
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-grid pf-m-gutter"
      >
        <div
          class="pf-l-grid__item pf-m-6-col pf-m-2-row"
        />
        <div
          class="pf-l-grid__item pf-m-6-col"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Review Information
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="test-result"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Test Result or Submission Type
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      class="pf-c-form-control"
                      disabled=""
                      id="test-result"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="test-date"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Effective Date
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      class="pf-c-form-control"
                      disabled=""
                      id="test-date"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-3"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Employee Information
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="profile-form-employee-id"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Person ID
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="profile-input-employee-id"
                      class="pf-c-form-control"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="profile-form-employee-name"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Name
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="profile-input-employee-name"
                      class="pf-c-form-control"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="profile-form-employee-email"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Email
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="profile-input-employee-email"
                      class="pf-c-form-control"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </article>
        </div>
      </div>
    </div>
    <div
      class="pf-c-card__footer"
    >
      <div
        class="pf-l-flex"
      >
        <div
          class="pf-m-align-left"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Back
          </button>
        </div>
        <div
          class="pf-m-align-right"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-primary"
            data-ouia-component-id="OUIA-Generated-Button-primary-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-c-button__icon pf-m-start"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                />
              </svg>
            </span>
            Confirm
          </button>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
